### DISCUSSION TITLE
Semester 4 Woche 5

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#45] [Student Exchange]
Hallo zusammen,

diese Woche ist vor Allem die ⚙️ Engine erweitert worden. 
Hier wurden Entities umprogrammiert, sodass diese sich fürs 👁️ Rendern eignen und bei denen Events registriert werden können (bspw. CollisionEvent oder MoveEvent). Außerdem wird ⌛ fleißig an dem behandeln dieser Events gearbeitet. 
Des Weiterem können jetzt 🖼️ Bilder als "Skybox" gerendert werden.
Hier könnt ihr den momentanen Stand erkennen. Insbesondere, dass 🐄 Entities auch "zwischen" 🧊 MapObjects platziert werden können.
![Screenshot_2024-05-06_113822](https://github.com/Christian-2003/CrInGE/assets/136824971/f5adcebc-7b1c-4fa2-b0c2-adddff95bf22)
Auch wurden die Klassen im Model-Package der Engine refactored, damit das 📦 Package nicht zu groß wird.

Das wäre dann auch schon alles für diese Woche. Im nächsten Blog werden wir euch weiter über das Projekt informieren.

**Comment by PetrusQn** [comment]
Hi CrInGE-Team,
Euer Blogeintrag gefällt mir gut und ich bin darauf gespannt, wie ihr die "Reaktion" auf solche Events umsetzen wollt. Zusätzlich halte ich die Entscheidung, den Code zu optimieren, für sehr gut und sinnvoll - weiter so 👍 
Ich freue mich auf den Blog der nächsten Woche 😽

**Comment by JanWilfert** [comment]
Hallo zusammen,
das sieht ja alles schon sehr gut aus. Gefällt mir auch sehr gut :) Mir stellt sich nur die Frage ob ihr für eure neuen Funktionen auch direkt automatische Tests geschrieben habt oder ob sich bei der Testabdeckung seit letzter Woche nichts geändert hat

**Comment by lukpr0** [comment]
Hallo Cringe-Team, uer Fortschritt sieht gut aus. Mich hätte noch interessiert, wenn ihr mehr auf das erwähnte Refactoring eingegangen wärt, inwiefern bzw. wie sehr hilft es der Package-Größe?
